sky/clouds/kubernetes.py

```diff
@@ -188,6 +189,12 @@ def existing_allowed_contexts(cls, silent: bool = False) -> List[str]:
             ctx for ctx in all_contexts if not ctx.startswith('ssh-')
         ]
 
+        allow_all_contexts = allowed_contexts is not None and (
+            allowed_contexts == 'all' or
+            env_options.Options.ALLOW_ALL_KUBERNETES_CONTEXTS.get())
```

Comment:
I believe this will be false if `allowed_contexts` is `None` even if the environment variable is set - did you mean
 ```suggestion
        allow_all_contexts = (allowed_contexts is not None and
            allowed_contexts == 'all') or
            env_options.Options.ALLOW_ALL_KUBERNETES_CONTEXTS.get()
```
?

Reply:
Looking at another place we have config and environment variable be able to control system behavior:

(skypilot_config.py)
```
        db_url = config.pop_nested(('db',), None)
        if db_url:
            os.environ[constants.ENV_VAR_DB_CONNECTION_URI] = db_url
```
We let config win over environment variable. Therefore, the behavior should be:
- If config is not specified and environment variable is not set, use active context.
- If config is not specified and environment variable is set, use all contexts.
- If config is specified, ignore environment variable.
Then, the correct code should be:
```
        allow_all_contexts = allowed_contexts == 'all' or (
            allowed_contexts is None and
            env_options.Options.ALLOW_ALL_KUBERNETES_CONTEXTS.get())
```

Reply:
I updated the logic for the `allow_all_contexts` : )